File: app/src/main/java/com/anokmik/databinding/component/TwoWayDataBindingComponent.java
package com.anokmik.databinding.component;

import android.databinding.BindingAdapter;
import android.databinding.DataBindingComponent;
import android.databinding.InverseBindingListener;

import com.anokmik.databinding.view.MaterialDesignPrimaryPaletteView;

import static com.anokmik.databinding.view.MaterialDesignPrimaryPaletteView.OnColorChangeListener;

public class TwoWayDataBindingComponent implements DataBindingComponent {

    @Override
    public TwoWayDataBindingComponent getTwoWayDataBindingComponent() {
        return this;
    }

    @BindingAdapter(value = {"onColorChange", "colorAttrChanged"}, requireAll = false)
    public void setColorListener(MaterialDesignPrimaryPaletteView view, final OnColorChangeListener listener, final InverseBindingListener colorChange) {
        if (colorChange == null) {
            view.setOnColorChangeListener(listener);
        } else {
            view.setOnColorChangeListener(new OnColorChangeListener() {
                @Override
                public void onColorChange(MaterialDesignPrimaryPaletteView view, int color) {
                    if (listener != null) {
                        listener.onColorChange(view, color);
                    }
                    colorChange.onChange();
                }
            });
        }
    }

    @BindingAdapter("color")
    public void setColor(MaterialDesignPrimaryPaletteView view, int color) {
        if (color != view.getColor()) {
            view.setColor(color);
        }
    }

}

File: app/src/main/kotlin/com/anokmik/databinding/component/TwoWayDataBindingComponent.kt
package com.anokmik.databinding.component

import android.databinding.BindingAdapter
import android.databinding.InverseBindingListener
import android.databinding.InverseBindingMethod
import android.databinding.InverseBindingMethods
import com.anokmik.databinding.view.MaterialDesignPrimaryPaletteView

@InverseBindingMethods(
        InverseBindingMethod(type = MaterialDesignPrimaryPaletteView::class, attribute = "color")
)
class TwoWayDataBindingComponent {

    @BindingAdapter(value = *arrayOf("onColorChange", "colorAttrChanged"), requireAll = false)
    fun setColorListener(view: MaterialDesignPrimaryPaletteView, listener: MaterialDesignPrimaryPaletteView.OnColorChangeListener?, colorChange: InverseBindingListener?) {
        if (colorChange == null) {
            view.listener = listener
        } else {
            view.listener = object : MaterialDesignPrimaryPaletteView.OnColorChangeListener {
                override fun onColorChange(view: MaterialDesignPrimaryPaletteView, color: Int) {
                    listener?.onColorChange(view, color)
                    colorChange.onChange()
                }
            }
        }
    }

    @BindingAdapter("color")
    fun setColor(view: MaterialDesignPrimaryPaletteView, color: Int) {
        if (color != view.color) {
            view.color = color
        }
    }

}